===================================================================
RCS file: /cvs/gnome/gtk+/gtk/gtkfilesystemunix.c,v
retrieving revision 1.77.2.2
retrieving revision 1.77.2.3
diff -u -r1.77.2.2 -r1.77.2.3
--- gtk+/gtk/gtkfilesystemunix.c	2006/09/04 20:49:21	1.77.2.2
+++ gtk+/gtk/gtkfilesystemunix.c	2006/09/29 14:17:01	1.77.2.3
@@ -255,6 +255,7 @@
 						   const char *mime_type);
 
 static gboolean execute_callbacks_idle (gpointer data);
+static void execute_callbacks (gpointer data);
 
 static gboolean fill_in_names     (GtkFileFolderUnix  *folder_unix,
 				   GError            **error);
@@ -443,7 +444,7 @@
       system_unix->execute_callbacks_idle_id = 0;
 
       /* call pending callbacks */
-      execute_callbacks_idle (system_unix);
+      execute_callbacks (system_unix);
     }
 
   G_OBJECT_CLASS (gtk_file_system_unix_parent_class)->dispose (object);
@@ -699,15 +700,13 @@
 
 
 
-static gboolean
-execute_callbacks_idle (gpointer data)
+static void
+execute_callbacks (gpointer data)
 {
   GSList *l;
   gboolean unref_file_system = TRUE;
   GtkFileSystemUnix *system_unix = GTK_FILE_SYSTEM_UNIX (data);
 
-  GDK_THREADS_ENTER ();
-
   if (!system_unix->execute_callbacks_idle_id)
     unref_file_system = FALSE;
   else
@@ -746,6 +745,14 @@
     g_object_unref (system_unix);
 
   system_unix->execute_callbacks_idle_id = 0;
+}
+
+static gboolean
+execute_callbacks_idle (gpointer data)
+{
+  GDK_THREADS_ENTER ();
+
+  execute_callbacks(data);
 
   GDK_THREADS_LEAVE ();
 
